CMakeLists.txt

```diff
@@ -339,6 +339,26 @@ opm_add_test(flow_onephase
   DEPENDS opmsimulators
   LIBRARIES opmsimulators)
 
+opm_add_test(flow_co2_diffusion
+  ONLY_COMPILE
+  DEFAULT_ENABLE_IF ${FLOW_VARIANTS_DEFAULT_ENABLE_IF}
+  SOURCES
+  flow/flow_co2_diffusion.cpp
+  $<TARGET_OBJECTS:moduleVersion>
+  EXE_NAME flow_co2_diffusion
+  DEPENDS opmsimulators
+  LIBRARIES opmsimulators)
+
+opm_add_test(flow_bo_diffusion
+  ONLY_COMPILE
+  DEFAULT_ENABLE_IF ${FLOW_VARIANTS_DEFAULT_ENABLE_IF}
+  SOURCES
+  flow/flow_bo_diffusion.cpp
+  $<TARGET_OBJECTS:moduleVersion>
+  EXE_NAME flow_bo_diffusion
+  DEPENDS opmsimulators
+  LIBRARIES opmsimulators)
```

Comment:
@akva2 : Is this the right way to enable new specialised simulators?  What, if anything, would we need change in the build system in order to incorporate these into the main `flow` executable if we wanted to?

Reply:
My reasoning for keeping them out of `flow ` at the moment is that I would like to test the performance drawback (if any) of always enabling diffusion. You will still need to provide the DIFFUSE keyword for it to have an effect, but compile time enabling flow may have computational overhead that needs investigation. If the overhead is small I would like to make a PR that enables diffusion by default. If the overhead is too large I would like to add the specialized version to  `flow ` in a later PR. The reason for adding them here as separate binaries is to simplify the testing.